### PR TITLE
feat: Add new ptn modules `azd/apim-api`

### LINKED ISSUE #1232: [Module Proposal]: `avm/ptn/azd/apim-api`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Bicep

### Module Classification?

Pattern Module

### Module Name

avm/ptn/azd/apim-api

### Module Details

Please add a Pattern Module named `avm/ptn/azd/apim-api`, this module will create the following two resources:

1. API Management Service.
2. Web Site.

For details, please refer to azd: 
https://github.com/Azure/azure-dev/blob/main/templates/todo/common/infra/bicep/app/apim-api.bicep

CC: @jongio

### Do you want to be the owner of this module?

No

### Module Owner's GitHub Username (handle)

jongio

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by jianingwang123**
## Description

Fixes [Azure/Azure-Verified-Modules/issues#1232](https://github.com/Azure/Azure-Verified-Modules/issues/1232)
## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.ptn.azd.apim-api](https://github.com/jianingwang123/bicep-registry-modules/actions/workflows/avm.ptn.azd.apim-api.yml)     |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio for notification.

### CODE REVIEW

**Comment by matebarabas** [comment]
@AlexanderSehr, @jianingwang123, good news: the MAR manifest has been updated with the required record(s) for this module. You can take the next steps accordingly. Thanks.

**Comment by matebarabas** [comment]
Please make sure the following disclaimer is added to the `readme.md` of this module - preferably somewhere close to the top: "*This module is not intended for broad, generic use, as it was designed to cater for the requirements of the AZD CLI product. Feature requests and bug fix requests are welcome if they support the development of the AZD CLI but may not be incorporated if they aim to make this module more generic than what it needs to be for its primary use case.*"

**Comment by v-xuto** [comment]
@AlexanderSehr Could you help review this PR?

File: avm/ptn/azd/apim-api/tests/e2e/defaults/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-apim-api-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('.')
+param apimServiceName string = '${namePrefix}-as-${serviceShort}001'
```
Comment (AlexanderSehr) [inline avm/ptn/azd/apim-api/tests/e2e/defaults/main.test.bicep:17]: please move into variable (that also doesn't need an empty description) 😉

File: avm/ptn/azd/apim-api/tests/e2e/defaults/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    Application_Type: 'web'
+    WorkspaceResourceId: logAnalyticsWorkspace.id
+  }
+}
+
+@description('The default hostname of the site.')
+output siteHostName string = 'https://${app.outputs.defaultHostname}'
+
+@description('The name of the API Management service.')
+output apimOutputName string = apimService.outputs.name
```
Comment (AlexanderSehr) [inline avm/ptn/azd/apim-api/tests/e2e/defaults/dependencies.bicep:84]:
```suggestion
output apimName string = apimService.outputs.name
```
I guess whether it's an output or not does not really matter to you in the `main.test.bicep`

**Comment by AlexanderSehr** [comment]
Hey @jianingwang123,
just one small other comment. Otherwise it appears we're good to go. The only other thing that needs fixing are the conflicts 💪

**Comment by jianingwang123** [comment]
@AlexanderSehr I have updated based on your comments, please re-review.

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Hey @jianingwang123,
please note that the module publishing **failed** because the static test failed. As per the [error](https://github.com/Azure/bicep-registry-modules/actions/runs/10792157555), it seems the generated files where to up to date when merging the PR.
To solve this we need a few things:
1. You need to create a new PR based on the latest code in main
2. In that PR you need to introduce a change to the `main.json` file. It does not matter what it is (e.g., it can be a change to the module's description or a parameter), but it must be something that tells the CI 'hey something that justifies a new module version changed'
3. You need to run the `Set-AVMModule` script again to make sure the files are up to date
4. You ideally also run the pipeline again so that we can ensure the tests wont' fail again

**Comment by jianingwang123** [comment]
> Hey @jianingwang123, please note that the module publishing **failed** because the static test failed. As per the [error](https://github.com/Azure/bicep-registry-modules/actions/runs/10792157555), it seems the generated files where to up to date when merging the PR. To solve this we need a few things:
> 
> 1. You need to create a new PR based on the latest code in main
> 2. In that PR you need to introduce a change to the `main.json` file. It does not matter what it is (e.g., it can be a change to the module's description or a parameter), but it must be something that tells the CI 'hey something that justifies a new module version changed'
> 3. You need to run the `Set-AVMModule` script again to make sure the files are up to date
> 4. You ideally also run the pipeline again so that we can ensure the tests wont' fail again

@AlexanderSehr  The [PR](https://github.com/Azure/bicep-registry-modules/pull/3256) for it is ready.

**Pull request merged**